### PR TITLE
feat(#2276): Migrate rest interfaces to spring

### PR DESCRIPTION
**Pull request by dominikriemer**
### Purpose

See #2276 

### Remarks

PR introduces (a) breaking change(s): no

PR introduces (a) deprecation(s): no

### CODE REVIEW

File: streampipes-rest-core-base/src/main/java/org/apache/streampipes/rest/core/base/impl/AbstractRestResource.java [hunk 1]
```diff
@@ -37,16 +36,59 @@
...
-
-import jakarta.ws.rs.core.Response;
+import org.springframework.http.ResponseEntity;
 
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URLDecoder;
 
-public abstract class AbstractRestResource extends AbstractSharedRestInterface {
```
Comment (bossenti) [inline streampipes-rest-core-base/src/main/java/org/apache/streampipes/rest/core/base/impl/AbstractRestResource.java:49]: Why do we remove abstraction here? Can we completely get rid of the `AbstractSharedRestInterface`?
If so, we can do this in a follow up task.

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/PipelineElementAsset.java [hunk 2]
```diff
@@ -22,47 +22,41 @@
...
+@RequestMapping("/api/v2/pe")
 public class PipelineElementAsset extends AbstractRestResource {
 
   private static final Logger LOG = LoggerFactory.getLogger(PipelineElementAsset.class);
 
-  @GET
-  @Path("{appId}/assets/icon")
-  @Produces("image/png")
-  public Response getIconAsset(@PathParam("appId") String appId) {
+  @GetMapping(path = "/{appId}/assets/icon", produces = "image/png")
```
Comment (bossenti) [inline streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/PipelineElementAsset.java:40]:
```suggestion
  @GetMapping(path = "/{appId}/assets/icon", produces = MediaType.IMAGE_PNG)
```

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/PipelineElementAsset.java [hunk 3]
```diff
@@ -22,47 +22,41 @@
...
     } catch (IOException e) {
       return fail();
     }
   }
 
-  @GET
-  @Path("{appId}/assets/{assetName}")
-  @Produces("image/png")
-  public Response getAsset(@PathParam("appId") String appId, @PathParam("assetName") String
+  @GetMapping(path = "/{appId}/assets/{assetName}", produces = "image/png")
```
Comment (bossenti) [inline streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/PipelineElementAsset.java:58]:
```suggestion
  @GetMapping(path = "/{appId}/assets/{assetName}", produces = MediaType.IMAGE_PNG)
```

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/admin/GeneralConfigurationResource.java [hunk 4]
```diff
@@ -67,23 +59,35 @@ public Response updateGeneralConfiguration(GeneralConfig config) {
     return ok();
   }
 
-  @GET
-  @Path("keys")
-  @Produces("multipart/mixed")
+  // TODO
```
Comment (bossenti) [inline streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/admin/GeneralConfigurationResource.java:62]:
```suggestion
```

File: streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/admin/GeneralConfigurationResource.java [hunk 5]
```diff
@@ -67,23 +59,35 @@ public Response updateGeneralConfiguration(GeneralConfig config) {
...
+  @GetMapping(path = "keys", produces = "multipart/mixed")
```
Comment (bossenti) [inline streampipes-rest/src/main/java/org/apache/streampipes/rest/impl/admin/GeneralConfigurationResource.java:63]:
```suggestion
  @GetMapping(path = "keys", produces = MediaType.MULTIPART_MIXED)
```

File: streampipes-service-core-minimal/src/main/java/org/apache/streampipes/service/core/minimal/StreamPipesCoreApplicationMinimal.java [hunk 6]
```diff
@@ -36,7 +35,7 @@
 
 @Configuration
 @EnableAutoConfiguration
-@Import({StreamPipesResourceConfig.class,
+@Import({//StreamPipesResourceConfig.class,
```
Comment (bossenti) [inline streampipes-service-core-minimal/src/main/java/org/apache/streampipes/service/core/minimal/StreamPipesCoreApplicationMinimal.java:38]: ?

**Review by bossenti — CHANGES_REQUESTED** [review]

**Review by bossenti — APPROVED** [review]

**Review by tenthe — APPROVED** [review]

**Pull request merged**